src/com/company/Deck.java

```diff
@@ -38,6 +38,15 @@ public void init(int numPacks) {
 
    public void shuffle() {
       // mixes up the cards with the help of the standard random number generator.
+      for(i = 0; i < topcard - 1; i++)
+      {
+        // select a random number between i and topcard
+        j = Random().nextInt(topcard - i) + i;
+        Card temp = new Card(cards[j]);
```

Comment:
Does this need to be a Card copy here? I think all we care about is saving the reference where `cards[j]` was, so that we can assign that reference to `cards[i]` later.